src/on.js

```diff
@@ -6,171 +6,229 @@
 	} else if (typeof exports === 'object') {
 		module.exports = factory();
 	} else {
-		root.returnExports = factory();
-		window.on = factory();
+		root.returnExports = window.on = factory();
 	}
 }(this, function () {
 	'use strict';
 
-	function hasWheelTest () {
-		var
-			isIE = navigator.userAgent.indexOf('Trident') > -1,
-			div = document.createElement('div');
-		return "onwheel" in div || "wheel" in div ||
-			(isIE && document.implementation.hasFeature("Events.wheel", "3.0")); // IE feature detection
-	}
+	// main function
 
-	var
-		INVALID_PROPS,
-		matches,
-		hasWheel = hasWheelTest(),
-		isWin = navigator.userAgent.indexOf('Windows') > -1,
-		FACTOR = isWin ? 10 : 0.1,
-		XLR8 = 0,
-		mouseWheelHandle;
+	function on (node, eventName, filter, handler) {
+		// normalize parameters
+		if (typeof node == 'string') {
+			node = getNodeById(node);
+		}
 
+		// prepare a callback
+		var callback = makeCallback(node, filter, handler);
 
-	['matches', 'matchesSelector', 'webkit', 'moz', 'ms', 'o'].some(function (name) {
-		if (name.length < 7) { // prefix
-			name += 'MatchesSelector';
+		// functional event
+		if (typeof eventName == 'function') {
```

Comment:
it is. `typeof` always returns a string. on the right side is a string constant. Do you expect `typeof` to return something else? Strictness makes sense only for non-equal types, like a number and a string &mdash; it will involve a conversion.